files/image_config/kubernetes/scripts/kube_join.sh

```diff
@@ -0,0 +1,235 @@
+#! /bin/bash
+
+# Redirect any stdout & stderr into logger
+#
+exec 1> >(logger -s -t kube_join_stdout) 2>&1
+
+API_SERVER=""
+CURL_FLAGS=""
+args_str=""
+
+ASYNC_RUN=false
+debug=false
+force=false
+test_only=false
+
+take_lock() {
+    # Check is Lock File exists, if not create it and set trap on exit
+    if { set -C; 2>/dev/null >/var/lock/kube_join.lock; }; then
+        trap "rm -f /var/lock/kube_join.lock" EXIT
+    else
+        echo "Lock file exists; exiting"
+        exit
+    fi
+
+}
+
+set_env() {
+    # Required when running as CRON job
+    export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+}
+
+log_info() {
+    logger -p local0.notice -t kube_join ${1}
+    if $debug; then
+        echo $1
+    fi
+}
+
+log_err() {
+    logger -p local0.error -t kube_join ${1}
+    if $debug; then
+        echo $1
+    fi
+}
+
+create_cron() {
+    log_info "Creating Cron job /usr/bin/kube_join.sh ${args_str}"
+    ( crontab -l | grep -v "kube_join" ; echo "*/10 * * * * /usr/bin/kube_join.sh ${args_str}" ) | crontab -
+}
+
+drop_cron() {
+    log_info "Dropping Cron job /usr/bin/kube_join.sh ${args_str}"
+    ( crontab -l | grep -v "kube_join" ) | crontab -
+}
+
+download_file() {
+    myfile=$(mktemp "${TMPDIR:-/tmp/}$(basename $0).XXXXXXXXXXXX")
+    if ! curl -f -s ${CURL_FLAGS} -o ${myfile} https://${API_SERVER}/admin.conf; then
+        log_err "Failed to download https://${API_SERVER}/admin.conf"
+        exit -1
+    fi
+    cp ${myfile} /etc/sonic/kube_admin.conf
+    rm -f ${myfile}
+    log_info "Downloaded kubeconfig into /etc/sonic/kube_admin.conf"
+}
+
+is_connected() {
+    pid=""
+    if [ -f /etc/sonic/kube_admin.conf ]; then
+        master=`kubectl --kubeconfig /etc/sonic/kube_admin.conf get nodes | grep master | cut -f1 -d' '`
+        if [ -n ${master} ]; then
+            pid=`kubectl --kubeconfig /etc/sonic/kube_admin.conf describe node ${master} | grep InternalIP | cut -f2- -d':' | tr -d '[:space:]'`
+        fi
+    fi
+    if [ "${pid}" == "${API_SERVER}" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+usage() {
+    cat << EOF
+Usage: [-i] [-a] [-d] [-s <API Server IP>] [-f]
+
+    -s -- IP address of API server"
+    -i -- Does insecure curl download of kubeconfig file 
+    -a -- Async mode
+    -d -- Debug mode
+    -f -- Force join
+    -t -- test only 
```

Comment:
What is the intention? Can you remove?

Reply:
They are all supported modes.
Async -- used, so the caller is not blocked
debug - Helps with verbose spew
force -- Force a reset & re-join
test -- Executes all pre-requisites, but not execute -- a easy way to validate your request

Reply:
I only mean `-t`. If you want to keep, please capitialize `Test`.

Reply:
Done